jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -471,6 +498,80 @@ default boolean rewind()
      */
     public interface Sink
     {
+        /**
+         * <p>Wraps the given {@link OutputStream} as a {@link Sink}.
+         * @param out The stream to wrap
+         * @return A sink wrapping the stream
+         */
+        static Sink from(OutputStream out)
+        {
+            return new Sink()
+            {
+                boolean closed;
+
+                @Override
+                public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
+                {
+                    if (closed)
+                    {
+                        callback.failed(new EOFException());
+                        return;
+                    }
+                    try
+                    {
+                        BufferUtil.writeTo(byteBuffer, out);
+                        if (last)
+                        {
+                            closed = true;
+                            out.close();
+                        }
+                        callback.succeeded();
+                    }
+                    catch (Throwable t)
+                    {
+                        callback.failed(t);
+                    }
+                }
+            };
+        }
+
+        /**
+         * <p>Wraps the given {@link OutputStream} as a {@link Sink}.
+         * @param channel The ByteChannel to wrap
+         * @return A sink wrapping the stream
+         */
+        static Sink from(ByteChannel channel)
+        {
+            return new Sink()
+            {
+                boolean closed;
+
+                @Override
+                public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
+                {
+                    if (closed)
+                    {
+                        callback.failed(new EOFException());
+                        return;
+                    }
+                    try
+                    {
+                        channel.write(byteBuffer);
```

Comment:
`ByteChannel.write(ByteBuffer)` does not guarantee that the whole buffer will be written, so you can't simply succeed the callback afterwards.

Reply:
oh good catch!